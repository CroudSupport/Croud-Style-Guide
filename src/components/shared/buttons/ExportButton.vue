<template>
    <button class="ui button" :class="{loading: downloading}" @click="exportData">
        <slot>Export</slot>
    </button>
</template>

<script>
    /**
     * Universal export button
     *
     * @example ./croud-export-button.md
     */
    export default {
        name: 'croud-export-button',

        props: {
           /**
            * export url
            */
            url: {
                type: String,
                required: true,
            },

           /**
            * local filename
            */
            filename: {
                type: String,
                default: 'Export.xlsx',
            },

           /**
            * mimetype
            */
            filetype: {
                type: String,
                default: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
            },

        },

        data() {
            return {
                downloading: false,
            }
        },

        methods: {
            exportData() {
                const xhr = new XMLHttpRequest()
                const self = this

                xhr.open('GET', this.url, true)
                xhr.responseType = 'blob'
                xhr.setRequestHeader('Authorization', `Bearer ${localStorage.getItem('jwt')}`)

                xhr.onload = function () {
                    if (this.status === 200) {
                        const link = document.createElement('a')
                        link.href = window.URL.createObjectURL(new Blob([this.response], { type: self.filetype }))
                        link.download = self.filename
                        link.click()
                        self.downloading = false
                    } else {
                        alert('Export failed')
                    }
                }

                this.downloading = true
                xhr.send()
            },
        },
    }
</script>
