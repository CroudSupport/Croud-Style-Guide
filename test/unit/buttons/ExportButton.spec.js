import Vue from 'vue'
import ExportButton from '../../../src/components/shared/buttons/ExportButton'

const Constructor = Vue.extend(ExportButton)

const vm = new Constructor().$mount()

describe('Export Button', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })
})
