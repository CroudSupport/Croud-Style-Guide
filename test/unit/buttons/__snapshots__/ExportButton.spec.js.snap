// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export Button should match the snapshot 1`] = `
<button
  class="ui button"
>
  Export
</button>
`;
